x/stake/endblock.go

```diff
@@ -12,8 +12,17 @@ import (
 
 func EndBlocker(ctx sdk.Context, k keeper.Keeper) (validatorUpdates []abci.ValidatorUpdate, completedUbds []types.UnbondingDelegation) {
 	// only change validator set in breath block after BEP159
-	var events sdk.Events
-	var csEvents sdk.Events
+	var (
+		events             sdk.Events
+		csEvents           sdk.Events
+		refundEvents       sdk.Events
+		closeChannelEvents sdk.Events
+	)
+	if sdk.IsUpgrade(sdk.BCFusionSecondHardFork) {
+		refundEvents = handleRefundCrossStake(ctx, k)
+		closeChannelEvents = closeSideChainChannels(ctx, k)
```

Comment:
I dont think staking module is a place to do `closeSideChainChannels`, the cross chain module is proper.